Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -119,6 +149,32 @@ endRideTransaction driverId bookingId ride mbFareParams mbRiderDetailsId merchan
     updateDriverWeeklyZscore ride rideDate weekStartDate weekEndDate driverWeeklyZscore ride.chargeableDistance
   DLoc.updateOnRideCacheForCancelledOrEndRide driverId merchantId
   SRide.clearCache $ cast driverId
+  unless (driverRideFee <= 0) $ do
+    maxShards <- asks (.maxShards)
+    isPendingPaymentJobScheduled <- getPendingPaymentCache driverFee.endTime
+    let pendingPaymentJobTs = diffUTCTime driverFee.endTime now
+    case isPendingPaymentJobScheduled of
+      Nothing -> do
+        Esq.runNoTransaction $
+          createJobIn @_ @'SendPaymentReminderToDriver pendingPaymentJobTs maxShards $
+            SendPaymentReminderToDriverJobData
+              { startTime = driverFee.startTime,
+                endTime = driverFee.endTime
+              }
+        setPendingPaymentCache driverFee.endTime
+      _ -> pure ()
+
+    let overduePaymentJobTs = diffUTCTime driverFee.payBy now
```

Comment:
same we can use `driverPaymentCycleDuration + driverPaymentCycleBuffer`

Reply:
This is the time FROM NOW, it is relative not absolute